Add tests for cap run

toniecrumley opened the pull request.

tsoelter commented:
I get this error from your instructions:

![Screen Shot 2025-05-01 at 10 14 11 AM](https://github.com/user-attachments/assets/8daa8479-3513-40e9-8ecc-23ac58909bf5)

tsoelter commented:
I got three failures (two were due to your blazerid). 

The third one is seen in this screenshot: ![Screen Shot 2025-05-01 at 10 37 35 AM](https://github.com/user-attachments/assets/5a592d8a-3074-4d11-a0fd-9360b67281a7)

toniecrumley commented:
> I got three failures (two were due to your blazerid in the cat logs command).
> 
> The third one is seen in this screenshot: ![Screen Shot 2025-05-01 at 10 37 35 AM](https://private-user-images.githubusercontent.com/62297985/439641169-5a592d8a-3074-4d11-a0fd-9360b67281a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxMTY3ODcsIm5iZiI6MTc0NjExNjQ4NywicGF0aCI6Ii82MjI5Nzk4NS80Mzk2NDExNjktNWE1OTJkOGEtMzA3NC00ZDExLWEwZmQtOTM2MGI2NzI4MWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAxVDE2MjEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZWJiNTYyYzk2NmFkNzk0MjE1ZDZiOWY1NjVkZTYwOGMwYWYyMzJhYmJjMWFlZGU2NWY1YTgxZTQwMmY1NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aK2G3DmSlPwHoOscwt9DGuYIjP6LMxYcZhVaWNeORjM)

I added a fix for these failures.  They were all a result of my username in the tests.

tsoelter reviewed (approved):
I get the expected outputs now @toniecrumley 🎉

lizzyjoan commented:
I got 28 failures, but really was just a repeated permission issue at /scratch/local/bin, so it seems like I might be missing something with my cap configuration, but I don't see anything right away when checking through the repo README.

Here's one failed test as an example:
``` ✗ cap md5 --slurm batch: All files in a folder
   (from function `stub' in file tests/commands/../../node_modules/bats-mock/stub.bash, line 19,
    in test file tests/commands/md5.bats, line 98)
     `stub mktemp " : echo '$temp_script'"' failed
   ln: failed to create symbolic link ‘/scratch/local/bin/mktemp’: Permission denied 
```

toniecrumley commented:
> I got 28 failures, but really was just a repeated permission issue at /scratch/local/bin, so it seems like I might be missing something with my cap configuration, but I don't see anything right away when checking through the repo README.
> 
> Here's one failed test as an example:
> 
> ```
>    (from function `stub' in file tests/commands/../../node_modules/bats-mock/stub.bash, line 19,
>     in test file tests/commands/md5.bats, line 98)
>      `stub mktemp " : echo '$temp_script'"' failed
>    ln: failed to create symbolic link ‘/scratch/local/bin/mktemp’: Permission denied 
> ```

@lizzyjoan Please try it again.  I deleted that bin directory.  When more than one developer runs tests within the timeframe that Cheaha cleans up temporary directories, these errors occur because Cheaha users seem to share a temp directory space.  I have fixed this problem in #28 by creating a unique temp directory for ever test run so you can also test there to make sure it keeps working.

adc0032 commented:
I'm getting a similar response as lizzy. but also see my comment on #28 if that is relevant.
<img width="1287" alt="Screenshot 2025-05-07 at 4 18 50 PM" src="https://github.com/user-attachments/assets/2b1f7f4c-e9b8-480c-b7c3-eebd54c1a1b2" />
